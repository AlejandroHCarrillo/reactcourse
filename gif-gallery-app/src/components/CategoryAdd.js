import React, { useState } from 'react'
import PropTypes from 'prop-types'
// setCategories es la funcion del componente padre
// la recibimos por medio de las PropTypes y la desestrucramos usando {}
export const CategoryAdd = ( { setCategories } ) => {
    const initialState = "";
    const [inputValue, setInputValue] = useState(initialState);

    const inputChangeHandler = (e) =>{
        // console.log(e.target.value);
        setInputValue(e.target.value);
    }

    const submitHandler = (e) =>{
        e.preventDefault();        
        
        if (inputValue.trim().length>0){
            setCategories( cats => [...cats, inputValue ]);
            setInputValue("");
            console.log("Enviado");
        }
    }

    return (
        <form onSubmit = { submitHandler }>
          <h2>{ inputValue }</h2>
          <input type="text"
                value = { inputValue }
                onChange = { inputChangeHandler }
          ></input>  
        </form>
    )
}

// En esta seccion usamos los PropTypes para hacer requerida 
// la funcion setCategories del componente padre
CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired
}