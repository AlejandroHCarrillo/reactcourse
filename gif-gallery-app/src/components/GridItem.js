import React from 'react'

export const GridItem = ( {img} ) => {
    // console.log(img);
    return (
        <div className="card">
                <img src={img.url} alt={img.title}></img>
                <p>{img.title}</p>
        </div>
    )
}
