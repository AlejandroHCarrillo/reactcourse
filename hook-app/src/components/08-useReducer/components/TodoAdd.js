import React from 'react'
import { useForm } from '../../../hooks/useForm';

export const TodoAdd = ({handleAddTodo}) => {

    const [ { description }, taskInputChangeHandler, reset ] = useForm({
        description: ''
    });


    const handleSubmit = (e) =>{
        e.preventDefault();
        if(description.length <= 0){
            return;
        }

        console.log("Submit Nueva tarea");

        const newTask = {
            id: new Date().getTime(),
            desc: description,
            done: false
        };

        // const action = {
        //     type: "add",
        //     payload: newTask
        // }

        handleAddTodo(newTask);

        reset();
    }

    return (
        <>
            <h4>Agregar tarea:</h4>
            <hr/>
            <form onSubmit={ handleSubmit }>
                <div className="d-grid gap-2">
                    <input type="text" 
                        className="form-control"
                        name="description"
                        placeholder="Escriba la tarea"
                        autoComplete="off"
                        value = { description }
                        onChange={ taskInputChangeHandler }                            
                        ></input>
                    <button type="submit"
                        className="btn btn-outline-success mt-1 btn-block"
                        >Agregar tarea +</button>
                </div>
            </form>            
        </>
    )
}
