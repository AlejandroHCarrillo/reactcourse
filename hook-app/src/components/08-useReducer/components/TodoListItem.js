import React from 'react'

export const TodoListItem = ({todo, index, handleDelete, handleToggle}) => {
    console.log('TodoListItem: ', todo);
    return (
        <li key={todo?.id}
        className="list-group-item"
        onClick={ () => handleToggle(todo.id) }
        >
        <p 
        className= { `${ todo?.done && 'complete' }` }
        
        >
        {index+1}. {todo?.desc}
        </p>

        <button className="btn btn-danger"
        onClick = { () => handleDelete(todo?.id) }
        >Eliminar</button>
        </li>
    )
}
