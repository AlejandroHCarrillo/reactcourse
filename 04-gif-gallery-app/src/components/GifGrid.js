import { useFetchGifs } from '../hooks/useFetchGifs';
import { GridItem } from './GridItem';
import PropTypes from 'prop-types'

export const GifGrid = ( { category } ) => {

    // console.log(category);
    const { data: images, loading} = useFetchGifs( category );

    // console.log( 'gifgrid', images, loading);
    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{ category }</h3>
            { loading && <p className=" animate__animated animate__flash">'Cargando...'</p> }
            <div className="card-grid">
                { images.map( img => (
                    // <GridItem key={img.id} img = { img } />
                    <GridItem key={img.id} title = { img.title } url = { img.url } />
                )) 
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}