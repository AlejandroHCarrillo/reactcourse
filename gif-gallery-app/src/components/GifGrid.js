// import React, { useEffect, useState } from 'react'
import React, { createRef } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
import { GridItem } from './GridItem';

const GifGrid = ( { category } ) => {

    // console.log(category);
    const { data: images, loading} = useFetchGifs( category );

    // console.log( images, loading);

    // const [images, setImages] = useState([]);
    // useEffect( () => {
    //     getGifs(category).then( setImages )
    // }, [category])
    
    return (
        <>
            <h3>{ category }</h3>
            { loading && 'Cargando...'  }
            <div className="card-grid">
                { images.map( img => (
                    <GridItem key={img.id} img={img} />
                )) 
                }
            </div>
        </>
    )
}

export default GifGrid
