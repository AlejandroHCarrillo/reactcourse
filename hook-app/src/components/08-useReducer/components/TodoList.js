import React from 'react'
import { TodoListItem } from './TodoListItem'

export const TodoList = ({todos=[], handleDelete, handleToggle}) => {
    console.log(todos);
    return (
          <ul className="list-group list-group-flush">
                        { 
                            todos.map( (t, i) =>  (
                                <TodoListItem  key={t.id} todo={ t } index={ i } handleDelete = { handleDelete }  handleToggle = { handleToggle }/>
                            ) )
                        }
                    </ul>   
    )
}
