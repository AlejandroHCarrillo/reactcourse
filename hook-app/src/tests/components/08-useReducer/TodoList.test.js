import React from 'react';
import { shallow } from 'enzyme';
import { demoTodos } from "../../fixtures/demoTodo"
import { TodoList } from "../../../components/08-useReducer/components/TodoList"



describe('Probando TodoList', () => {

    const handleDelete = jest.fn(); // funcion dummy usando jest para cumplir el parametro requerido
    const handleToggle = jest.fn(); // funcion dummy usando jest para cumplir el parametro requerido

    const wrapper = shallow(
        <TodoList todos={demoTodos} 
                  handleDelete={handleDelete} 
                  handleToggle={handleToggle}
        />);

    test('should mostrar el componente', () => {
        // console.log(wrapper.html());
        expect(wrapper).toMatchSnapshot();     
    });

    test('should de contener los elementos del demoTodo', () => {

        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        console.log(wrapper.find('TodoListItem').at(0).props() );

        // revisa que este la funcion handleDelete
        expect( wrapper.find('TodoListItem').at(0).prop('handleDelete') ).toEqual(expect.any(Function));
        // revisa que este la funcion handleToggle
        expect( wrapper.find('TodoListItem').at(1).prop('handleToggle') ).toEqual(expect.any(Function));
    });
    
});
