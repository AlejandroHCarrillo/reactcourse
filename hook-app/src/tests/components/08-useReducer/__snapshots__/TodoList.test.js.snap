// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando TodoList should mostrar el componente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender react",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
